packages/runtime-vapor/src/componentSlots.ts

```diff
@@ -39,50 +49,45 @@ export const initSlots = (
   }
 
   if (dynamicSlots) {
+    slots = shallowReactive(slots)
     const dynamicSlotKeys: Record<string, true> = {}
-    baseWatch(
-      () => {
-        const _dynamicSlots = dynamicSlots()
-        for (let i = 0; i < _dynamicSlots.length; i++) {
-          const slot = _dynamicSlots[i]
-          // array of dynamic slot generated by <template v-for="..." #[...]>
-          if (isArray(slot)) {
-            for (let j = 0; j < slot.length; j++) {
-              slots[slot[j].name] = withCtx(slot[j].fn)
-              dynamicSlotKeys[slot[j].name] = true
-            }
-          } else if (slot) {
-            // conditional single slot generated by <template v-if="..." #foo>
-            slots[slot.name] = withCtx(
-              slot.key
-                ? (...args: any[]) => {
-                    const res = slot.fn(...args)
-                    // attach branch key so each conditional branch is considered a
-                    // different fragment
-                    if (res) (res as any).key = slot.key
-                    return res
-                  }
-                : slot.fn,
-            )
-            dynamicSlotKeys[slot.name] = true
+    renderEffect(() => {
```

Comment:
Why change here? `slots` can be accessed by public API `useSlots`. So we need to update it before effects from user-land. Just like `attrs`

https://github.com/vuejs/core-vapor/blob/6c99c7a89d18cfd6db2214ad1c72408354e2ccde/packages/runtime-vapor/src/componentProps.ts#L125

Reply:
When making the modification, I did not take this situation into account, but I have now corrected it.

Maybe we can add a unit test for this feature later?